website/pages/docs/secrets/databases/custom.mdx

```diff
@@ -12,81 +12,165 @@ description: |-
+
+	// Close attempts to close the underlying database connection that was
+	// established by the backend.
 	Close() error
 }
 ```
 
-You'll notice the first parameter to a number of those functions is a
-`Statements` struct. This struct is used to pass the Role's configured
-statements to the plugin on function call. The struct is defined as:
+Each of the request and response objects can also be found in `sdk/database/dbplugin/v5/database.go`.
 
-```go
-type Statements struct {
-	Creation   []string
-	Revocation []string
-	Rollback   []string
-	Renewal    []string
-}
-```
+In each of the requests, you will see a `Statements` object. This object represents the set of
+commands to run for that particular operation. For the `NewUser` function, this is a set of
+commands to create the user (and often set permissions for that user). These statements are
+from the following fields in the API:
 
-It is up to your plugin to replace the `{{name}}`, `{{password}}`, and
-`{{expiration}}` in these statements with the proper values.
+| API Argument               | Request Object                                     |
+| -------------------------- | -------------------------------------------------- |
+| `creation_statements`      | `NewUserRequest.Statements.Commands`               |
+| `revocation_statements`    | `DeleteUserRequest.Statements.Commands`            |
+| `rollback_statements`      | `NewUserRequest.RollbackStatements.Commands`       |
+| `renew_statements`         | `UpdateUserRequest.Expiration.Statements.Commands` |
+| `rotation_statements`      | `UpdateUserRequest.Password.Statements.Commands`   |
+| `root_rotation_statements` | `UpdateUserRequest.Password.Statements.Commands`   |
 
-The `Initialize` function is passed a map of keys to values, this data is what the
+In many of the built in plugins, they replace `{{name}}` (or `{{username}}`), `{{password}}`,
+and/or `{{expiration}}` with the associated values. It is up to your plugin to perform these
+string replacements. There is a helper function located in `sdk/database/helper/dbutil`
+called `QueryHelper` that assists in doing this string replacement. You are not required to
+use it, but it will make your plugin's behavior consistent with the built in plugins.
+
+The `InitializeRequest` object contains a map of keys to values. This data is what the
 user specified as the configuration for the plugin. Your plugin should use this
-data to make connections to the database. It is also passed a boolean value
-specifying whether or not your plugin should return an error if it is unable to
-connect to the database.
+data to make connections to the database. The response object contains a similar configuration
+map. The response object should contain the configuration map that should be saved within Vault.
+This allows the plugin to manipulate the configuration prior to saving it.
+
+It is also passed a boolean value indicating if your plugin should initialize a connection to
```

Comment:
I think we should rephrase this sentence a bit and directly mention `IntializeRequest.VerifyConnection` as the boolean value that controls this behavior.

Reply:
:+1: